consensus/src/experimental/tests/commit_phase_tests.rs

```diff
@@ -0,0 +1,383 @@
+    waypoint::Waypoint,
+};
+use futures::{executor::block_on, future::FutureExt};
+use network::protocols::network::{Event, NewNetworkSender};
+use safety_rules::{PersistentSafetyStorage, SafetyRulesManager};
+use std::collections::BTreeMap;
+
+use consensus_types::experimental::{commit_decision::CommitDecision, commit_vote::CommitVote};
+use diem_types::block_info::BlockInfo;
+use tokio::runtime::Runtime;
+
+fn prepare_commit_phase() -> (
+    Sender<(Vec<ExecutedBlock>, LedgerInfoWithSignatures)>,
+    Sender<ConsensusMsg>,
+    Receiver<(Vec<Block>, LedgerInfoWithSignatures)>,
+    Receiver<Event<ConsensusMsg>>,
+    Arc<Mutex<MetricsSafetyRules>>,
+    Vec<ValidatorSigner>,
+    Arc<OrderingStateComputer>,
+    ValidatorVerifier,
+    Runtime,
+) {
+    let num_nodes = 1;
+
+    // constants
+    let channel_size = 30;
+    let back_pressure = Arc::new(AtomicU64::new(0));
+
+    // environment setup
+    let runtime = consensus_runtime();
+    let _executor = runtime.handle().clone();
+    let (signers, validators) = random_validator_verifier(num_nodes, None, false);
+    let validator_set = (&validators).into();
+    let signer = &signers[0];
+
+    let waypoint =
+        Waypoint::new_epoch_boundary(&LedgerInfo::mock_genesis(Some(validator_set))).unwrap();
+
+    let safety_storage = PersistentSafetyStorage::initialize(
+        Storage::from(diem_secure_storage::InMemoryStorage::new()),
+        signer.author(),
+        signer.private_key().clone(),
+        Ed25519PrivateKey::generate_for_testing(),
+        waypoint,
+        true,
+    );
+    let safety_rules_manager = SafetyRulesManager::new_local(safety_storage, false, false, true);
+
+    let (_initial_data, storage) = MockStorage::start_for_testing((&validators).into());
+    let epoch_state = EpochState {
+        epoch: 1,
+        verifier: storage.get_validator_set().into(),
+    };
+    let validators = epoch_state.verifier.clone();
+    let (network_reqs_tx, _network_reqs_rx) = diem_channel::new(QueueStyle::FIFO, 8, None);
+    let (connection_reqs_tx, _) = diem_channel::new(QueueStyle::FIFO, 8, None);
+
+    let network_sender = ConsensusNetworkSender::new(
+        PeerManagerRequestSender::new(network_reqs_tx),
+        ConnectionRequestSender::new(connection_reqs_tx),
+    );
+    let author = signer.author();
+
+    let (self_sender, self_receiver) = channel::new_test(1000);
+    let network = NetworkSender::new(author, network_sender, self_sender, validators);
+
+    let (commit_result_sender, commit_result_receiver) =
+        channel::new_test::<(Vec<Block>, LedgerInfoWithSignatures)>(channel_size);
+    let state_computer = Arc::new(OrderingStateComputer::new(commit_result_sender));
+
+    let mut safety_rules = MetricsSafetyRules::new(safety_rules_manager.client(), storage);
+    safety_rules.perform_initialize().unwrap();
+
+    let safety_rules_container = Arc::new(Mutex::new(safety_rules));
+
+    // setting up channels
+    let (sender_comm, receiver_comm) =
```

Comment:
it's idiomatic to to call the sender, receiver (tx, rx). we can name it (commit_tx, commit_rx), similarly for the msg (msg_tx, msg_rx)

Reply:
resolved